src/include/txn.i

```diff
@@ -716,20 +716,23 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
     *updp = NULL;
     for (; upd != NULL; upd = upd->next) {
         /* Skip reserved place-holders, they're never visible. */
-        if (upd->type != WT_UPDATE_RESERVE) {
-            upd_visible = __wt_txn_upd_visible_type(session, upd);
-            if (upd_visible == WT_VISIBLE_TRUE)
-                break;
-            if (upd_visible == WT_VISIBLE_PREPARE)
-                return (WT_PREPARE_CONFLICT);
+        if (upd->type == WT_UPDATE_RESERVE)
+            continue;
+        upd_visible = __wt_txn_upd_visible_type(session, upd);
+        if (upd_visible == WT_VISIBLE_TRUE) {
+            if (upd->type != WT_UPDATE_BIRTHMARK)
+                *updp = upd;
+            return (0);
         }
+        if (upd_visible == WT_VISIBLE_PREPARE)
+            return (WT_PREPARE_CONFLICT);
     }
 
-    /* Cannot find the update in memory. Try lookaside. */
-    if (upd == NULL && __wt_page_las_active(session, cbt->ref))
+    /* If there's no visible update in memory, check the history file. */
```

Comment:
Better to use lookaside file instead of history file as of now. We will change the comment also when we change the file name.
Update the comment from "If there's no visible update in memory" to "If there's no visible update in data store".

Reply:
I think data store is inaccurate here as we are reading the update chain. Changed to update chain.